errors.go

```diff
@@ -70,13 +70,19 @@ func newCellNameToCoordinatesError(cell string, err error) error {
 	return fmt.Errorf("cannot convert cell %q to coordinates: %v", cell, err)
 }
 
+// newColumnNumberError defined the error message on receiving invalid column
```

Comment:
I agree with you. We need to declare two exported error variable `ErrMinColumnNumber` and `ErrMaxColumnNumber`, remove the `newColumnNumberError` function, and rename the `TotalColumns` to `MaxColumns`.

```go
    // ErrMaxColumnNumber defined the error message on receive a column
    // number exceeds maximum limit.
    ErrMaxColumnNumber = fmt.Errorf("column number exceeds the maximum limit %d", MaxColumns)
    // ErrMinColumnNumber defined the error message on receive a column
    // number less than the minimum limit.
    ErrMinColumnNumber = fmt.Errorf("column number less than the minimum limit %d", MinColumns)
```

Reply:
Hm… while having 2 error types is the most accurate, in normal usage just knowing that the column number is wrong (as a type) is sufficient - imho. And therefore makes checking for library user easier - if they want to handle the column number error.  
We can add extra detail using a func like in the original PR version to tell if greater than MAX or less than MIN:

```golang
ErrColumnNumber = errors.New("invalid column number")

func newColumnNumberError(count int) error {
    var detail string
	if count < MinColumns {
		detail = fmt.Sprintf("%d is less than minimum limit of 1 columns", count)
	} else if count > TotalColumns {
		detail = fmt.Sprintf("%d exceeds maximum limit of %d columns", count, TotalColumns)
	} else {
		detail = "internal error: incorrect error usage"
	}
	return fmt.Errorf("%w: %s", ErrColumnNumber, detail)
}
```

And if the library user wants further error inspection, they can check against exported `MaxColumns`/`MinColumns` constants to do special handling on their side, if any at all.  
2 errors make checking more elaborate…

Or do you want to stick with the 2 error types?

Reply:
I suggest using two error types and letting the user know which kind of error directly in details. Do you thinks so?

Reply:
Do you mean sth. like this?

```golang
ErrColumnNumber = errors.New("column number out of bounds")
ErrMaxColumnNumber = fmt.Errorf("column number exceeds the maximum limit %d", MaxColumns)
ErrMinColumnNumber = fmt.Errorf("column number less than the minimum limit %d", MinColumns)

func newColumnNumberError(count int) error {
    var detail error
	if count < MinColumns {
		detail = ErrMinColumnNumber
	} else {
		detail = ErrMaxColumnNumber
	}
	return fmt.Errorf("%w: %w", ErrColumnNumber, detail)
}
```

If you did think of this, then yes, that would be nice. But sadly impossible, as golang currently does not allow multi-wrapping errors like this…

If you meant sth. else, then I'm lost